frontend/ui/screensaver.lua

```diff
@@ -377,6 +382,38 @@ function Screensaver:show(event, fallback_message)
     end
 end
 
+function Screensaver:expandSpecial(message, fallback)
+    -- Expand special character sequences in given message. Use fallback string if there is no document instance
+    -- %p percentage read, with percent sign
+    -- %c current page
+    -- %t total pages
+    -- %dt document title
+
+    local ret = message
+
+    local lastfile = G_reader_settings:readSetting("lastfile")
+    local doc = DocumentRegistry:openDocument(lastfile)
+    local instance = require("apps/reader/readerui"):_getRunningInstance()
+    if instance ~= nil then
+        local currentpage = instance.view.state.page
+        ret = string.gsub(ret, "%%c", currentpage)
+
+        local totalpages = doc:getPageCount()
+        ret = string.gsub(ret, "%%t", totalpages)
+
+        local percent = math.floor(((currentpage * 100) / totalpages) + 0.5)
```

Comment:
We also have our own `Math.round()` function which, admittedly, does exactly this. But the advantage is it's easier to search the source to see where it's used. :-)

https://github.com/koreader/koreader/blob/93b6335421dcc63321e1aac3526b005aa10da52a/frontend/optmath.lua#L19-L21

Reply:
OK cool, I'll use that then.

Reply:
Oops, I forgot about that (am on my phone). Minor follow-up PR if you want? Doesn't matter too much.

Reply:
Might as well finish it off properly.